zeus/device/cpu/rapl.py

```diff
@@ -0,0 +1,180 @@
+        try:
+            with open(os.path.join(path, "name"), "r") as name_file:
+                self.name: str = name_file.read().strip()
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError("Error reading package name") from err
+        try:
+            with open(self.energy_uj_path) as energy_file:
+                self.last_energy = int(float(energy_file.read().strip()) / 10**3)
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError("Error reading package energy") from err
+        try:
+            with open(
+                os.path.join(path, "max_energy_range_uj"), "r"
+            ) as max_energy_file:
+                self.max_energy_range_uj = int(
+                    float(max_energy_file.read().strip()) / 10**3
+                )
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError(
+                "Error reading package max energy range"
+            ) from err
+
+    def __str__(self) -> str:
+        """Return a string representation of the RAPL file object."""
+        return f"Path: {self.path}\nEnergy_uj_path: {self.energy_uj_path}\nName: {self.name}\
+        \nLast_energy: {self.last_energy}\nMax_energy: {self.max_energy_range_uj}"
+
+    def _read(self) -> int:
+        """Read the current energy value from the energy_uj file.
+
+        Returns:
+            int: The current energy value in millijoules.
+        """
+        with open(self.energy_uj_path) as energy_file:
+            self.last_energy = int(float(energy_file.read().strip()) / 10**3)
+        return self.last_energy
+
+    def _read_delta(self) -> int:
+        """Read the delta energy value since the last read.
+
+        Returns:
+            int: The delta energy value in millijoules.
+        """
+        last_energy: int = self.last_energy
+        new_energy: int = self._read()
+        if new_energy < last_energy:
+            return new_energy + self.max_energy_range_uj - last_energy
+        return new_energy - last_energy
+
+
+class RAPLCPU(cpu_common.CPU):
+    """Control a Single Intel CPU using RAPL interface."""
+
+    def __init__(self, cpu_index: int) -> None:
+        """Initialize the Intel CPU with a specified index."""
+        super().__init__(cpu_index)
+        self._get_powerzone()
+
+    _exception_map = {
+        FileNotFoundError: cpu_common.ZeusCPUNotFoundError,
+        PermissionError: cpu_common.ZeusCPUNoPermissionError,
+        OSError: cpu_common.ZeusCPUInitError,
+    }
+
+    def _get_powerzone(self) -> None:
+        self.path = os.path.join(RAPL_DIR, f"intel-rapl:{self.cpu_index}")
+        self.rapl_file: RAPLFile = RAPLFile(self.path)
+        self.dram: RAPLFile | None = None
+        for dir in os.listdir(self.path):
+            if "intel-rapl" in dir:
+                rapl_file = RAPLFile(os.path.join(self.path, dir))
+                if rapl_file.name == "dram":
+                    self.dram = rapl_file
```

Comment:
`RAPLFile`'s constructor will throw a `ZeusRAPLFileInitError` if `intel-rapl:0:0` doesn't have `name` or `energy_uj`, but I think that's not right. If somehow `intel-rapl:0:0` is strangely configured but `intel-rapl:0` looks fine, I think we should just `warnings.warn` the resulting `ZeusRAPLFileInitError` error, assume that only CPU measurement is supported, and proceed normally.